cloud/store/recover.go

```diff
@@ -0,0 +1,192 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package store
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/raft"
+	raftbolt "github.com/hashicorp/raft-boltdb/v2"
+	"github.com/vmihailenco/msgpack"
+)
+
+// Peer represents information about a peer in the Raft cluster
+type Peer struct {
+	Address  string `msgpack:"Address" json:"address"`
+	ID       string `msgpack:"ID" json:"id"`
+	Suffrage int64  `msgpack:"Suffrage" json:"-"`
+	NonVoter bool   `msgpack:"-" json:"non_voter"`
+}
+
+// genPeersFileFromBolt checks if there was an existing raft.db and has persisted config
+// so it generates peers.jon file to be used for raft.RecoverCluster to recover the cluster on start.
+func (st *Store) genPeersFileFromBolt(raftDBPath, peerFilePath string) error {
+	if _, err := os.Stat(raftDBPath); err != nil {
+		st.log.Error("can not generate peers file from existed bolt db", err)
+		// if files doesn't exist we bail out early
+		return nil
+	}
+	bdb, err := raftbolt.NewBoltStore(raftDBPath)
+	if err != nil {
+		return err
+	}
+	defer bdb.Close()
+
+	f, err := bdb.FirstIndex()
+	if err != nil {
+		st.log.Error("can not generate peers file from existed bolt db", err)
+		return nil
+	}
+	l, err := bdb.LastIndex()
+	if err != nil {
+		st.log.Error("can not generate peers file from existed bolt db", err)
+		return nil
+	}
+
+	st.log.Info(fmt.Sprintf("found previous config at %s, recovery will be initiated", raftDBPath))
```

Comment:
Does not return an error if the index does not exist

Reply:
> If a node restarts before joining the leader, an empty raft.db will exist. The node will not be able to recover thereafter. In this case, we need to ignore the empty raft.db file

if the index doesn't exists because in case of `raft.db`  is empty it will return nil